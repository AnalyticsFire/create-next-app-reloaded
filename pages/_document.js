// ./pages/_document.js
import React from 'react';
import Document, { Head, Main, NextScript } from 'next/document';
import htmlescape from 'htmlescape';
import config from 'config';

export default class MyDocument extends Document {
  render() {
    return (
      <html>
        <Head>
          <link rel="stylesheet" href="/_next/static/style.css" />
        </Head>
        <body>
          <Main />
          <script
            // eslint-disable-next-line react/no-danger
            dangerouslySetInnerHTML={{
              __html: `
                __CONFIG__ = ${htmlescape(config)}
              `,
            }}
          />
          <NextScript />
        </body>
      </html>
    );
  }
}
