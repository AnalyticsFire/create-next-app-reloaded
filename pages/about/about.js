import Layout from '@root/components/layout';

const About = () => (
  <Layout title="About" className="about-page">
    <h1>About</h1>
  </Layout>
);

export default About;
