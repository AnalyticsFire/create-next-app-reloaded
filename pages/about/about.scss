.about-page {
  h1 {
    font-size: $titleSize;
  }
}