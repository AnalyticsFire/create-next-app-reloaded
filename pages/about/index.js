import Component, { About } from './about';

export { About };
export default Component;
