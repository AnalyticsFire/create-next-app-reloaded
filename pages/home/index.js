import Component, { Home } from './home';

export { Home };
export default Component;
