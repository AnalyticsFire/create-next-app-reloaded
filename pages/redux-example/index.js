import Component, { ReduxExample } from './reduxExample';

export { ReduxExample };
export default Component;
