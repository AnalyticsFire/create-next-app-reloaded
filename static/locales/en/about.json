{
  "pageTitle": "AnalyticsFire - About me",
  "title": "Analytics Fire started with a simple goal",
  "goal": "\"To help innovative smart device manufacturers bring complex system products to market.\"",
  "how": "In order to do this, we've developed a laser-like focus on all the specific skills necessary to design, build, and maintain complex IoT applications, while simultaneously developing a technology-agnostic practice that requires mastery of both the newest modern technologies (eg: MQTT, Cassandra, and React.js) as well as stable platform technologies already in place throughout the enterprise (eg: SCADA, PI Server, Salesforce Platform, and others)."
}