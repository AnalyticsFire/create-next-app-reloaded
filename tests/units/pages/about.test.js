/* eslint-disable no-undef */

import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { About } from '@root/pages/about';

const defaultComponent = <About t={() => {}} />;

test('About is rendered', () => {
  expect(toJson(shallow(defaultComponent))).toMatchSnapshot();
});

/* eslint-enable no-undef */
