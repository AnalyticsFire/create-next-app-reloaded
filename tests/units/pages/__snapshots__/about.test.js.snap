// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About is rendered 1`] = `
<Layout
  className="about-page"
>
  <div
    className="container pt-4 pb-4"
  >
    <h1
      className="title text-center pb-3"
    />
    <blockquote
      className="blockquote text-center"
    />
    <p
      className="text-center"
    />
  </div>
</Layout>
`;
